core/capabilities/webapi/target/target.go

```diff
@@ -0,0 +1,169 @@
+	return capabilityInfo, nil
+}
+
+func getMessageID(req capabilities.CapabilityRequest) (string, error) {
+	if err := validation.ValidateWorkflowOrExecutionID(req.Metadata.WorkflowID); err != nil {
+		return "", fmt.Errorf("workflow ID is invalid: %w", err)
+	}
+	if err := validation.ValidateWorkflowOrExecutionID(req.Metadata.WorkflowExecutionID); err != nil {
+		return "", fmt.Errorf("workflow execution ID is invalid: %w", err)
+	}
+	messageID := []string{
+		req.Metadata.WorkflowID,
+		req.Metadata.WorkflowExecutionID,
+		webapicapabilities.MethodWebAPITarget,
+	}
+	return strings.Join(messageID, "/"), nil
+}
+
+func (c *Capability) Execute(ctx context.Context, req capabilities.CapabilityRequest) (capabilities.CapabilityResponse, error) {
+	c.lggr.Debugw("executing http target", "capabilityRequest", req)
+
+	var input Input
+	err := req.Inputs.UnwrapTo(&input)
+	if err != nil {
+		return capabilities.CapabilityResponse{}, err
+	}
+
+	var workflowCfg WorkflowConfig
+	err = req.Config.UnwrapTo(&workflowCfg)
+	if err != nil {
+		return capabilities.CapabilityResponse{}, err
+	}
+
+	messageID, err := getMessageID(req)
+	if err != nil {
+		return capabilities.CapabilityResponse{}, err
+	}
+
+	c.registeredWorkflowsMu.RLock()
+	if _, ok := c.registeredWorkflows[req.Metadata.WorkflowID]; !ok {
+		c.registeredWorkflowsMu.RUnlock()
+		return capabilities.CapabilityResponse{}, fmt.Errorf("workflow is not registered: %v", req.Metadata.WorkflowID)
+	}
+	c.registeredWorkflowsMu.RUnlock()
+
+	payload := webapicapabilities.TargetRequestPayload{
+		URL:       input.URL,
+		Method:    input.Method,
+		Headers:   input.Headers,
+		Body:      []byte(input.Body),
+		TimeoutMs: workflowCfg.TimeoutMs,
+	}
+
+	payloadBytes, err := json.Marshal(payload)
+	if err != nil {
+		return capabilities.CapabilityResponse{}, err
+	}
+
+	switch workflowCfg.Schedule {
+	case SingleNode:
+		// blocking call to handle single node request. waits for response from gateway
+		resp, err := c.connectorHandler.HandleSingleNodeRequest(ctx, messageID, payloadBytes)
+		if err != nil {
+			return capabilities.CapabilityResponse{}, err
+		}
+		c.lggr.Debugw("received gateway response", "resp", resp)
+		var payload webapicapabilities.TargetResponsePayload
+		err = json.Unmarshal(resp.Body.Payload, &payload)
+		if err != nil {
+			return capabilities.CapabilityResponse{}, err
+		}
+
+		// TODO: check target response format and fields
+		values, err := values.NewMap(map[string]any{
+			"statusCode": payload.StatusCode,
+			"headers":    payload.Headers,
+			"body":       string(payload.Body),
+		})
+		if err != nil {
+			return capabilities.CapabilityResponse{}, err
+		}
+		return capabilities.CapabilityResponse{
+			Value: values,
+		}, nil
+	default:
+		return capabilities.CapabilityResponse{}, fmt.Errorf("unsupported schedule: %v", workflowCfg.Schedule)
+	}
+}
+
+func (c *Capability) RegisterToWorkflow(ctx context.Context, req capabilities.RegisterToWorkflowRequest) error {
```

Comment:
I was thinking about this too, but it may be a skeleton for rate limiting / retries. If this is true, might be good to add a TODO comment just for context.

Reply:
got it. if workflow engine provides that guarantee, i can remove this. Probably need to store retry config here later and can leave a TODO for that